src/main/java/edu/cornell/kfs/module/purap/businessobject/IWantAccount.java

```diff
@@ -207,6 +209,46 @@ public void setUseAmountOrPercent(String useAmountOrPercent) {
     }
 
     /**
+     * Returns the amount-or-percent value if configured to use amounts, zero otherwise.
+     * 
+     * @see org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntrySourceDetail#getAmount()
+     */
+    @Override
+    public KualiDecimal getAmount() {
+        return CUPurapConstants.AMOUNT.equals(getUseAmountOrPercent()) ? getAmountOrPercent() : KualiDecimal.ZERO;
+    }
+
+    // Only here to comply with the GeneralLedgerPendingEntrySourceDetail interface.
+    @Override
+    public String getBalanceTypeCode() {
+        return null;
+    }
+
+    // Only here to comply with the GeneralLedgerPendingEntrySourceDetail interface.
+    @Override
+    public String getFinancialDocumentLineDescription() {
+        return null;
```

Comment:
Should this throw a Run time exception or something of the like instead of just returning NULL?